pkg/cmd/proxy/kubectl/http-proxy-server.go

```diff
@@ -0,0 +1,171 @@
+// Copyright Contributors to the Open Cluster Management project
+package kubectl
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	grpccredentials "google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/keepalive"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+	konnectivity "sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client"
+)
+
+type httpProxyServer struct {
```

Comment:
can you elaborate why we need this local proxy server?

Reply:
Sure, the call chain is like:

clusteradm -> kubectl -> 【 local httpProxyServer】 -> port-forward proxyServer -> proxyServer on hub -> ...

The usage of this local httpProxyServer is used to translate HTTP requests(from kubectl) to GRPC requests.